### PR TITLE
Create new trip notification screen

### PR DESCRIPTION
**Pull request by annhit29**
In this pull request,
1) I create the notification screen (CreateNotification.kt) for the administrator of a trip to announce with the use of CreateNotificationViewModel.
2) Both the title and the description of the notifications are mandatory to be filled, with the title size cannot exceed 55 characters.
Note: With the use of `trim()` function, we ensure that the mandatory fields cannot be filled with only whitespace characters.

This pull request is from the issue #154.

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+                  // Logic to create notification
+                  val tripNotification =
+                      TripNotification(
+                          notificationId = "", // modified by database
+                          userId = "", // modified by database
+                          title = title,
+                          userName = "", // modified by database
+                          description = description,
+                          timestamp = LocalDateTime.now())
+                  // if successful, call onSuccess(), otherwise onFailure()
```
Comment (Exino329) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt:113]: For the moment, it is not necessary to define onFailure() and onSuccess() case behaviour, for the moment we assume that after adding the notification in the view Model, we will just navigate back to the notification screen. 
We can assume that if the add of the notif fails, we also navigate back to the Notification screen. (We can change the behaviour lated if required.

File: app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt
Comment (Exino329) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt:1]: Good class overall ! Few things could be probably improved  : 
- It is not necessary to define 3 arguments `onSuccess` `onFailure` `onCancel` because `onCancel` `onSuccess`  have the same purpose : navigating back to the notification screen. It is a good idea to think about a failure `onFailure` but for the moment it is simplier to assume that we don't check for failure from the data base when adding a notification and simply also go back to the notification screen.
- It could be nice to leave an errorMessage to notify which  field entry is not valid in addition to the red borders to provide more information to the user.
- The textfield for the description could be a bit larger (vertically) since we have space available for this screen. 
- It is better if the content (i.e. the two textfields and the button) is closer from the cancel button (i.e. put them higher) to not leave a feeling to the user that the screen is "empty".

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/CreateSuggestion.kt
Comment (Exino329) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/CreateSuggestion.kt:1]: This modification is not related to the current issue. It is not a problem however be careful to only make changes that are addressing the related issue/PR. I just leave this comment as a warning and don't worry I also already did this kind of things :')

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/CreateNotificationViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+import com.github.se.wanderpals.model.repository.TripsRepository
+import kotlinx.coroutines.launch
+
+/**
+ * ViewModel for creating notifications. This ViewModel is responsible for adding a notification to
+ * a trip.
+ */
+open class CreateNotificationViewModel(tripsRepository: TripsRepository) : ViewModel() {
+  private val _tripsRepository = tripsRepository
+
```
Comment (Exino329) [inline app/src/main/java/com/github/se/wanderpals/model/viewmodel/CreateNotificationViewModel.kt:15]: It could be nice to leave a comment to describe what the function does to improve code readability.

**Review by Exino329 — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt (hunk 1 above)
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt:113]: Okay! And if required, we can still modify it later :D

File: app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/CreateSuggestion.kt
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/CreateSuggestion.kt:1]: Thank you for the reminder!

File: app/src/main/java/com/github/se/wanderpals/model/viewmodel/CreateNotificationViewModel.kt (hunk 2 above)
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/model/viewmodel/CreateNotificationViewModel.kt:15]: Yes, I added it! Please have a check :D

File: app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/notification/CreateNotification.kt:1]: Thank you for your precious review!
1) I changed both `onSuccess` and `onCancel` to `onNavigationBack`. I kept `onFailure` and we discuss what we will do later for this.
2) Yes, I added them: "Title cannot be empty" and "Description cannot be empty" for the trip notification title and description respectively. Moreover, the word counter and the outlinedTextField of the title change the color from red back to gray when there is a letter.  The outlinedTextField of the description works in the same way.
3) Yes, I changed the OutlinedTextField of the description to `height(450.dp)`.
4) Yes, this is done by enlarging the OutlinedTextField of the description :D

**Review by Exino329 — APPROVED** [review]
